### PR TITLE
WIP: Rename store to place

### PR DESCRIPTION
**Pull request by nicksellen**
## What does this PR do?

Renames store to place. Works with the backend in https://github.com/yunity/karrot-backend/pull/585

The frontend appears to be the same as far as the user is concerned. Says "store" everywhere still. Or should do anyway.

## Links to related issues

https://github.com/yunity/karrot-frontend/issues/1127

## Checklist

- [ ] added a test, or explain why one is not needed/possible...
- [ ] no unrelated changes
- [ ] asked someone for a code review
- [ ] joined #karrot-dev channel at https://slackin.yunity.org
- [ ] added an entry to CHANGELOG.md (description, pull request link, username(s))

### CODE REVIEW

File: src/base/app.js [hunk 1]
```diff
@@ -58,6 +58,6 @@ export default async function initApp () {
 
   // makes it easier to remote debug vue in cordova
   // for example to access vuex, type this into the console
-  // window.vueRoot.$store.state
+  // window.vueRoot.$place.state
```
Comment (tiltec) [inline src/base/app.js]: This shouldn't be changed.

File: src/applications/pages/ApplicationForm.vue [hunk 2]
```diff
@@ -12,7 +12,7 @@ export default connect({
     apply: 'groupApplications/apply',
   },
   methodsToEvents: {
-    cancel: (store, groupId) => {
+    cancel: (place, groupId) => {
```
Comment (tiltec) [inline src/applications/pages/ApplicationForm.vue]: This should be `datastore` or `_` as it's not used.

File: src/base/i18n.js [hunk 3]
```diff
@@ -50,7 +50,7 @@ for (const locale of Object.values(locales)) {
 
 const i18n = new VueI18n({
   // Just need to include 'en' here as it is the fallback locale
-  // All other locales are loaded on demand in store/plugins/i18n
+  // All other locales are loaded on demand in place/plugins/i18n
```
Comment (tiltec) [inline src/base/i18n.js]: This should be `base/datastore/i18nPlugin` (leftover from module refactor)

File: src/base/routes/main.js [hunk 4]
```diff
@@ -248,46 +248,46 @@ export default [
         },
       },
       {
-        name: 'stores',
-        path: 'store',
+        name: 'places',
+        path: 'place',
```
Comment (tiltec) [inline src/base/routes/main.js]: I'd like to keep public URLs stable, so maybe add a redirect/alias from `store`?

File: src/base/routes/main.js [hunk 5]
```diff
@@ -248,46 +248,46 @@ export default [
...
           breadcrumbs: [
-            { translation: 'GROUP.STORES', route: { name: 'stores' } },
+            { translation: 'GROUP.STORES', route: { name: 'places' } },
           ],
         },
-        component: StoreList,
+        component: PlaceList,
       },
       {
         name: 'storeCreate',
```
Comment (tiltec) [inline src/base/routes/main.js]: route name should be `placeCreate`

File: src/base/routes/main.js [hunk 6]
```diff
@@ -248,46 +248,46 @@ export default [
...
           ],
         },
-        component: StoreCreate,
+        component: PlaceCreate,
       },
       {
-        redirect: '/group/:groupId/store/:storeId/wall',
-        path: 'store/:storeId',
+        redirect: '/group/:groupId/place/:storeId/wall',
+        path: 'place/:storeId',
```
Comment (tiltec) [inline src/base/routes/main.js]: needs redirect
should be `:placeId`

File: src/base/routes/main.js [hunk 7]
```diff
@@ -248,46 +248,46 @@ export default [
...
+            { translation: 'GROUP.STORES', route: { name: 'places' } },
           ],
         },
-        component: StoreList,
+        component: PlaceList,
       },
       {
         name: 'storeCreate',
-        path: 'store/create',
+        path: 'place/create',
```
Comment (tiltec) [inline src/base/routes/main.js]: needs redirect

File: src/base/routes/main.js [hunk 8]
```diff
@@ -339,7 +339,7 @@ export default [
               ],
               beforeEnter: 'history/fetch',
             },
-            component: StoreHistory,
+            component: PlaceHistory,
           },
           {
             name: 'storeEdit',
```
Comment (tiltec) [inline src/base/routes/main.js]: should be `placeEdit`

File: src/feedback/components/FeedbackItem.vue [hunk 9]
```diff
@@ -107,11 +107,11 @@ export default {
       return this.feedback && this.feedback.createdAt
     },
     storeName () {
-      const { about: { store: { name } = {} } = {} } = this.feedback
+      const { about: { place: { name } = {} } = {} } = this.feedback
       return name
     },
     storeId () {
```
Comment (tiltec) [inline src/feedback/components/FeedbackItem.vue]: should be `placeId`

File: src/feedback/components/FeedbackItem.vue [hunk 10]
```diff
@@ -107,11 +107,11 @@ export default {
       return this.feedback && this.feedback.createdAt
     },
     storeName () {
```
Comment (tiltec) [inline src/feedback/components/FeedbackItem.vue]: should be `placeName`

File: src/feedback/datastore/feedback.js [hunk 11]
```diff
@@ -46,7 +46,7 @@ export default {
   actions: {
     ...withMeta({
       async fetch ({ dispatch }, { groupId, storeId }) {
```
Comment (tiltec) [inline src/feedback/datastore/feedback.js]: should be `placeId`

File: src/pickups/components/PickupList.vue [hunk 12]
```diff
@@ -9,7 +9,7 @@
           v-for="pickup in pickups"
           :key="pickup.id"
           :pickup="pickup"
-          :store-link="storeLink"
+          :place-link="storeLink"
```
Comment (tiltec) [inline src/pickups/components/PickupList.vue]: should be `placeLink`

**Comment by nicksellen** [comment]
Thanks for the review!

I didn't address how to handle actually deploying these changes yet either, as people will have the old app installed which won't work with the new backend. A few options to think about soon.

**Comment by tiltec** [comment]
For deployment I agree, just change everything over.

What would you change in the translations, rename the message ids? I have the feeling we could delay this a bit when considering #1118, which would change all translations anyways.

File: src/places/pages/Layout.vue [hunk 13]
```diff
@@ -17,30 +17,30 @@
...
   methods: {
     restore () {
-      this.$datastore.dispatch('stores/save', { id: this.store.id, status: 'created' })
+      this.$datastore.dispatch('places/save', { id: this.place.id, status: 'created' })
     },
   },
 }
 </script>
 
 <style scoped lang="stylus">
```
Comment (tiltec) [inline src/places/pages/Layout.vue]: these styles (`store-banner`) are actually unused. maybe a quick unrelated change..?

File: src/subscriptions/firebase.js [hunk 14]
```diff
@@ -35,7 +35,7 @@ export async function initializeMessaging () {
 /*
  * When we disable push notifications we need to remove the service workers.
  * If we remove them immediately, then re-enable push notifications then
- * firebase messsaging library gets upset. So we just store whether we want them removed
+ * firebase messsaging library gets upset. So we just place whether we want them removed
```
Comment (tiltec) [inline src/subscriptions/firebase.js]: Don't. just don't...

File: src/topbar/datastore/breadcrumbs.spec.js [hunk 15]
```diff
@@ -4,12 +4,12 @@ jest.mock('@/base/i18n', () => ({ t: mockTranslate }))
 import { createDatastore } from '>/helpers'
 
 describe('breadcrumbs', () => {
-  let storeMocks
+  let placeMocks
```
Comment (tiltec) [inline src/topbar/datastore/breadcrumbs.spec.js]: `datastoreMocks`

File: src/utils/datastore/helpers.js [hunk 16]
```diff
@@ -1,6 +1,6 @@
 /**
- * Store independent helpers
- * If a helper depends on the vuex store, put it in storeHelpers.js
+ * Place independent helpers
```
Comment (tiltec) [inline src/utils/datastore/helpers.js]:
```suggestion
 * Datastore independent helpers
```

File: src/utils/datastore/helpers.js [hunk 17]
```diff
@@ -1,6 +1,6 @@
...
+ * If a helper depends on the vuex place, put it in storeHelpers.js
```
Comment (tiltec) [inline src/utils/datastore/helpers.js]:
```suggestion
 * If a helper depends on the vuex datastore, put it in storeHelpers.js
```

File: src/utils/datastore/storeHelpers.js [hunk 18]
```diff
@@ -1,5 +1,5 @@
 /**
- * Store dependent helpers
+ * Place dependent helpers
```
Comment (tiltec) [inline src/utils/datastore/storeHelpers.js]:
```suggestion
 * Datastore dependent helpers
```

**Comment by nicksellen** [comment]
Closing in favour of https://github.com/yunity/karrot-frontend/pull/1267
